### DISCUSSION TITLE
Error in use of sampler LCA

### DISCUSSION THREAD

**Discussion by uzuuzu20002000** [vladmandic/sdnext#2547] [Q&A] [answered]
Thank you vladmandic for all your help. I am impressed with the beautiful images you are able to generate.

I have built the environment in paperspace and am trying to generate it with Diffusers.
When I place LORA in the prompt and run it with LCA as the sampler, I get the following error.

**KDiffusionSampler.__init__() takes 3 positional arguments but 4 were given**

Is there some setting that is wrong?

※If I set the sampler to DEIS, etc., I can generate normally

I would be happy to receive your advice.

**Comment by vladmandic** [comment]
This sounds like an issue, so please create one and provide all details, especially log showing the error.

**Reply by uzuuzu20002000** [reply]
Thank you for contacting us. 

18:23:30-979178 INFO     Applying free-u: b1=1.2 b2=1.4 s1=0.9 s2=0.2           
18:23:34-223765 INFO     Applying LoRA: ['sdxl_lcm'] patch=0.00 load=3.24       
18:23:34-225339 ERROR    Exception: KDiffusionSampler.__init__() takes 3        
                         positional arguments but 4 were given                  
18:23:34-227010 ERROR    Arguments: args=('task(brsq5n31593xwl7)', "A woman's   
                         bare hands open a doorknob; the background is a school 
                         hallway,  <lora:sdxl_lcm:1.0>", 'Watermark,', [], 20,  
                         7, 7, True, False, False, 1, 1, 6, 6, 0.7, 1, -1.0,    
                         -1.0, 0, 0, 0, 1024, 1024, True, 0.5, 2, 'ESRGAN       
                         4x-UltraMix_Balanced', True, 20, 0, 0, 5, 0.8, '', '', 
                         [], 0, 'from modules.processing import                 
                         process_images\n\np.width = 768\np.height =            
                         768\np.batch_size = 2\np.steps = 10\n\nreturn          
                         process_images(p)', 2, False, False, 'positive',       
                         'comma', 0, False, False, '', 0, '', [], 0, '', [], 0, 
                         '', [], False, True, False, False, False, False, 0,    
                         False, False, {'ad_model': 'face_yolov8n.pt',          
                         'ad_prompt': '', 'ad_negative_prompt': '',             
                         'ad_confidence': 0.3, 'ad_mask_k_largest': 0,          
                         'ad_mask_min_ratio': 0, 'ad_mask_max_ratio': 1,        
                         'ad_x_offset': 0, 'ad_y_offset': 0, 'ad_dilate_erode': 
                         4, 'ad_mask_merge_invert': 'None', 'ad_mask_blur': 4,  
                         'ad_denoising_strength': 0.4, 'ad_inpaint_only_masked':
                         True, 'ad_inpaint_only_masked_padding': 32,            
                         'ad_use_inpaint_width_height': False,                  
                         'ad_inpaint_width': 512, 'ad_inpaint_height': 512,     
                         'ad_use_steps': False, 'ad_steps': 28,                 
                         'ad_use_cfg_scale': False, 'ad_cfg_scale': 7,          
                         'ad_use_checkpoint': False, 'ad_checkpoint': 'Use same 
                         checkpoint', 'ad_use_vae': False, 'ad_vae': 'Use same  
                         VAE', 'ad_use_sampler': False, 'ad_sampler': 'Default',
                         'ad_use_noise_multiplier': False,                      
                         'ad_noise_multiplier': 1, 'ad_use_clip_skip': False,   
                         'ad_clip_skip': 1, 'ad_restore_face': False,           
                         'ad_controlnet_model': 'None', 'ad_controlnet_module': 
                         'None', 'ad_controlnet_weight': 1,                     
                         'ad_controlnet_guidance_start': 0,                     
                         'ad_controlnet_guidance_end': 1, 'is_api': ()},        
                         {'ad_model': 'None', 'ad_prompt': '',                  
                         'ad_negative_prompt': '', 'ad_confidence': 0.3,        
                         'ad_mask_k_largest': 0, 'ad_mask_min_ratio': 0,        
                         'ad_mask_max_ratio': 1, 'ad_x_offset': 0,              
                         'ad_y_offset': 0, 'ad_dilate_erode': 4,                
                         'ad_mask_merge_invert': 'None', 'ad_mask_blur': 4,     
                         'ad_denoising_strength': 0.4, 'ad_inpaint_only_masked':
                         True, 'ad_inpaint_only_masked_padding': 32,            
                         'ad_use_inpaint_width_height': False,                  
                         'ad_inpaint_width': 512, 'ad_inpaint_height': 512,     
                         'ad_use_steps': False, 'ad_steps': 28,                 
                         'ad_use_cfg_scale': False, 'ad_cfg_scale': 7,          
                         'ad_use_checkpoint': False, 'ad_checkpoint': 'Use same 
                         checkpoint', 'ad_use_vae': False, 'ad_vae': 'Use same  
                         VAE', 'ad_use_sampler': False, 'ad_sampler': 'Default',
                         'ad_use_noise_multiplier': False,                      
                         'ad_noise_multiplier': 1, 'ad_use_clip_skip': False,   
                         'ad_clip_skip': 1, 'ad_restore_face': False,           
                         'ad_controlnet_model': 'None', 'ad_controlnet_module': 
                         'None', 'ad_controlnet_weight': 1,                     
                         'ad_controlnet_guidance_start': 0,                     
                         'ad_controlnet_guidance_end': 1, 'is_api': ()},        
                         {'ad_model': 'None', 'ad_prompt': '',                  
                         'ad_negative_prompt': '', 'ad_confidence': 0.3,        
                         'ad_mask_k_largest': 0, 'ad_mask_min_ratio': 0,        
                         'ad_mask_max_ratio': 1, 'ad_x_offset': 0,              
                         'ad_y_offset': 0, 'ad_dilate_erode': 4,                
                         'ad_mask_merge_invert': 'None', 'ad_mask_blur': 4,     
                         'ad_denoising_strength': 0.4, 'ad_inpaint_only_masked':
                         True, 'ad_inpaint_only_masked_padding': 32,            
                         'ad_use_inpaint_width_height': False,                  
                         'ad_inpaint_width': 512, 'ad_inpaint_height': 512,     
                         'ad_use_steps': False, 'ad_steps': 28,                 
                         'ad_use_cfg_scale': False, 'ad_cfg_scale': 7,          
                         'ad_use_checkpoint': False, 'ad_checkpoint': 'Use same 
                         checkpoint', 'ad_use_vae': False, 'ad_vae': 'Use same  
                         VAE', 'ad_use_sampler': False, 'ad_sampler': 'Default',
                         'ad_use_noise_multiplier': False,                      
                         'ad_noise_multiplier': 1, 'ad_use_clip_skip': False,   
                         'ad_clip_skip': 1, 'ad_restore_face': False,           
                         'ad_controlnet_model': 'None', 'ad_controlnet_module': 
                         'None', 'ad_controlnet_weight': 1,                     
                         'ad_controlnet_guidance_start': 0,                     
                         'ad_controlnet_guidance_end': 1, 'is_api': ()}, True,  
                         False, 1, False, False, False, 1.1, 1.5, 100, 0.7,     
                         False, False, True, False, False, 0,                   
                         'Gustavosta/MagicPrompt-Stable-Diffusion', '', False,  
                         7, 100, 'Constant', 0, 'Constant', 0, 4, True, 'MEAN', 
                         'AD', 1, <scripts.animatediff_ui.AnimateDiffProcess    
                         object at 0x7f1a9c94e6b0>, False, 0, 1, 0, 'Version 2',
                         1.2, 0.9, 0, 0.5, 0, 1, 1.4, 0.2, 0, 0.5, 0, 1, 1, 1,  
                         0, 0.5, 0, 1) kwargs={}                                
18:23:34-237751 ERROR    gradio call: TypeError                                 
╭───────────────────── Traceback (most recent call last) ──────────────────────╮
│ /notebooks/automatic/modules/call_queue.py:34 in f                           │
│                                                                              │
│   33 │   │   │   try:                                                        │
│ ❱ 34 │   │   │   │   res = func(*args, **kwargs)                             │
│   35 │   │   │   │   progress.record_results(id_task, res)                   │
│                                                                              │
│ /notebooks/automatic/modules/txt2img.py:66 in txt2img                        │
│                                                                              │
│   65 │   if processed is None:                                               │
│ ❱ 66 │   │   processed = processing.process_images(p)                        │
│   67 │   p.close()                                                           │
│                                                                              │
│ /notebooks/automatic/modules/processing.py:739 in process_images             │
│                                                                              │
│    738 │   │   │   with context_hypertile_vae(p), context_hypertile_unet(p): │
│ ❱  739 │   │   │   │   res = process_images_inner(p)                         │
│    740 │   finally:                                                          │
│                                                                              │
│ /notebooks/automatic/modules/processing.py:888 in process_images_inner       │
│                                                                              │
│    887 │   │   │   │   from modules.processing_diffusers import process_diff │
│ ❱  888 │   │   │   │   x_samples_ddim = process_diffusers(p, p.seeds, p.prom │
│    889 │   │   │   else:                                                     │
│                                                                              │
│ /notebooks/automatic/modules/processing_diffusers.py:388 in                  │
│ process_diffusers                                                            │
│                                                                              │
│   387 │   recompile_model()                                                  │
│ ❱ 388 │   update_sampler(shared.sd_model)                                    │
│   389 │   p.extra_generation_params['Pipeline'] = shared.sd_model.__class__. │
│                                                                              │
│ /notebooks/automatic/modules/processing_diffusers.py:383 in update_sampler   │
│                                                                              │
│   382 │   │   │   │   sampler = sd_samplers.all_samplers_map.get("UniPC")    │
│ ❱ 383 │   │   │   sd_samplers.create_sampler(sampler.name, sd_model)         │
│   384 │   │   │   # TODO extra_generation_params add sampler options         │
│                                                                              │
│ /notebooks/automatic/modules/sd_samplers.py:59 in create_sampler             │
│                                                                              │
│   58 │   elif shared.backend == shared.Backend.DIFFUSERS:                    │
│ ❱ 59 │   │   sampler = config.constructor(model)                             │
│   60 │   │   if not hasattr(model, 'scheduler_config'):                      │
│                                                                              │
│ /notebooks/automatic/extensions/sd-webui-animatediff/scripts/animatediff_lcm │
│ .py:128 in <lambda>                                                          │
│                                                                              │
│   127 │   │   samplers_data_lcm = [                                          │
│ ❱ 128 │   │   │   sd_samplers_common.SamplerData(label, lambda model, funcna │
│   129 │   │   │   for label, funcname, aliases, options in samplers_lcm      │
│                                                                              │
│ /notebooks/automatic/extensions/sd-webui-animatediff/scripts/animatediff_lcm │
│ .py:109 in __init__                                                          │
│                                                                              │
│   108 │   def __init__(self, funcname, sd_model, options=None):              │
│ ❱ 109 │   │   super().__init__(funcname, sd_model, options)                  │
│   110 │   │   self.model_wrap_cfg = CFGDenoiserLCM(self)                     │
╰──────────────────────────────────────────────────────────────────────────────╯
TypeError: KDiffusionSampler.__init__() takes 3 positional arguments but 4 were 
given

**Reply by uzuuzu20002000** [reply]
![lcm](https://github.com/vladmandic/automatic/assets/130671689/28cc2b4a-5514-45cd-b314-ee1d50067b68)

**Reply by vladmandic** [reply]
animatediff is not supported with diffusers backend. disty already provided answer yesterday.

**Reply by uzuuzu20002000** [reply]
The animatediff is installed but not used. 
Does this mean that uninstalling it will make it usable? 
Sorry for my lack of understanding.

**Comment by Disty0** [comment, marked as answer]
Use LCM with Diffusers backend.
That LCM is from Animatediff extension.

**Reply by uzuuzu20002000** [reply]
Does this mean that it can only be used in animation? I apologize if I am greatly mistaken.

**Reply by vladmandic** [reply]
there are TWO LCM samplers. REAL one, created by SD.Next when in Diffusers mode and whatever AnimateDiff creates  - which should be called "AnimateDiff LCM" instead of just "LCM". choose correct one.

**Reply by uzuuzu20002000** [reply]
Thank you very much.
I understand very well.
I will check and try.

And thank you so much.
I always support and appreciate your support.

**Reply by uzuuzu20002000** [reply]
I just realized that I did not communicate it properly.
Disty0 thank you so much!
